### PR TITLE
Update for pixi.js 4

### PR DESCRIPTION
**Pull request by dan335**
Probably not compatible with pixi.js < 4.

### CODE REVIEW

**Comment by marekventur** [comment]
You could check PIXI.VERSION and make sure it works for both 3 and 4 - I'd happy to merge it then

**Comment by dan335** [comment]
Just checked it with 3.  Does not work so I guess not yet.

**Comment by marekventur** [comment]
You could check the current PIXI version and switch between both approaches. Even if PIXI v4 would be released we will have to maintain backwards compatibility to v3 anyway.

**Comment by dan335** [comment]
Good point.  Updated it, works with both 3 and 4 now.

File: index.js [hunk 1]
```diff
@@ -66,25 +66,44 @@ module.exports = function (PIXI)
...
+                    var crop;
+                    if (Number(PIXI.VERSION.charAt(0)) >= 4) {
+                        crop = new PIXI.Rectangle(
+                            0,
+                            0,
+                            sprite.dimension.w / resolution,
+                            sprite.dimension.h / resolution
+                        );
+                    } else {
+                        crop = frame.clone();
```
Comment (marekventur) [inline index.js:79]: this needs to be 

``` js
crop = new PIXI.Rectangle(
    0,
    0,
    sprite.dimension.w / resolution,
    sprite.dimension.h / resolution
);
```

File: index.js [hunk 2]
```diff
@@ -66,25 +66,44 @@ module.exports = function (PIXI)
...
+                        } else {
+                            trim = new PIXI.Rectangle(
+                                sprite.trim.x / resolution,
+                                sprite.trim.y / resolution,
+                                frame.width,
+                                frame.height
+                            );
+
+                            crop.width = sprite.trim.w / resolution;
+                            crop.height = sprite.trim.h / resolution;
```
Comment (marekventur) [inline index.js:105]: same here, could you keep the old commands?

``` js
trim = new PIXI.Rectangle(
    sprite.trim.x / resolution,
    sprite.trim.y / resolution,
    sprite.trim.w / resolution,
    sprite.trim.h / resolution
);

frame.width = sprite.trim.w / resolution;
frame.height = sprite.trim.h / resolution;
```

Comment (dan335) [inline index.js:105]: I think the meaning of crop has changed.  It's now called "orig" not crop and it's the full rectangle of the original sprite so x,y are 0,0 and w,h are the full dimensions.

File: index.js (hunk 1 above)
Comment (dan335) [inline index.js:79]: Not sure what you mean.

File: index.js (hunk 2 above)
Comment (dan335) [inline index.js:105]: https://github.com/pixijs/pixi.js/blob/dev/src/core/textures/Texture.js#L22

Comment (marekventur) [inline index.js:105]: yes, but that's for >4, not for v3 (e.g. your else branch)? If the goal is backward compatibility nothing should change for pixi 3 users.

Comment (dan335) [inline index.js:105]: Same here.  The original code for v3 is.

```
trim = new PIXI.Rectangle(
    sprite.trim.x / resolution,
    sprite.trim.y / resolution,
    frame.width,
    frame.height
);

crop.width = sprite.trim.w / resolution;
crop.height = sprite.trim.h / resolution;
```

File: index.js (hunk 1 above)
Comment (dan335) [inline index.js:79]: The original code for v3 is 

```
crop = frame.clone();
```

Comment (marekventur) [inline index.js:79]: oh, I'm so sorry. I've just looked at your last commit and not the whole diff. I apologise for the confusion. Let me quickly test it and then I'll merge and release.

Comment (dan335) [inline index.js:79]: 👍

**Comment by marekventur** [comment]
@dan335 thanks for doing this!

**Comment by dan335** [comment]
Thanks for making it!  Made things much easer.

**Comment by marekventur** [comment]
released as https://github.com/Gamevy/pixi-packer-parser/tree/v1.2.0

**Pull request merged**